cmd/installer/main.go

```diff
@@ -85,17 +92,46 @@ func install(logger log.DebugLogger) error {
 			return err
 		}
 	}
-	if err := copyLogs(); err != nil {
+	if err := copyLogs(logFlusher); err != nil {
 		return fmt.Errorf("error copying logs: %s", err)
 	}
-	syscall.Sync()
-	time.Sleep(time.Second)
 	if err := unmountStorage(logger); err != nil {
 		return fmt.Errorf("error unmounting: %s", err)
 	}
 	return nil
 }
 
+func printAndWait(initialTimeoutString, waitTimeoutString string,
+	waitGroup *sync.WaitGroup, logger log.Logger) {
+	initialTimeout, _ := time.ParseDuration(initialTimeoutString)
+	if initialTimeout < time.Second {
+		initialTimeout = time.Second
+		initialTimeoutString = "1s"
+	}
+	logger.Printf("waiting %s before rebooting\n", initialTimeoutString)
+	time.Sleep(initialTimeout - time.Second)
+	waitChannel := make(chan struct{})
+	go func() {
+		waitGroup.Wait()
+		waitChannel <- struct{}{}
+	}()
+	timer := time.NewTimer(time.Second)
+	select {
+	case <-timer.C:
+	case <-waitChannel:
+		return
+	}
+	logger.Printf(
+		"waiting %s for remote shells to terminate before rebooting\n",
+		waitTimeoutString)
+	waitTimeout, _ := time.ParseDuration(waitTimeoutString)
+	timer.Reset(waitTimeout)
+	select {
+	case <-timer.C:
```

Comment:
maybe print something if you got a timeout?

Reply:
It will reboot right after this anyway. Also, such a message doesn't seem that interesting.